lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # Car behaviour
+  class Car
+    attr_reader :color, :model, :year, :current_speed
+
+    def initialize(car)
+      @year = car[:year] || 2015
+      @color = car[:color] || 'purple'
+      @model = car[:model] || 'skyline'
+      @current_speed = 0
+    end
+
+    def push_break(x)
+      @current_speed -= x if @current_speed >= x
```

Comment:
Names like `x`, `y`, `j` or `k` are not acceptable in Ruby.
what is `x`?

and you can try guard-pattern for instead after conditions
```ruby
def push_break(x)
  return if @current_speed <= x

  @current_speed -= x
end
```